app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt

```diff
@@ -80,6 +135,20 @@ class BackgroundServices(
         }
     }
 
+    private val accountObserver = object : AccountObserver {
+        override fun onAuthenticationProblems() {}
+
+        override fun onProfileUpdated(profile: Profile) {}
+
+        override fun onLoggedOut() {
+            pushService.stop()
```

Comment:
Can this affect any other code that interacts with Firebase? E.g. leanplum.

Reply:
Yes it can, but we've made Leanplum dependant on this working first since Firebase/Push are tightly coupled together.

There is a check on this observer not being added if we don't have push keys so that should be a graceful handling of the service.

Reply:
I should clarify in the ticket. We're making Firebase work properly with Send Tab first, before we decide to add support for LeanPlum into Fenix.